cpp/src/components/connectivity.cu

```diff
@@ -16,27 +19,38 @@
 //
 /**
  * @brief Compute connected components. 
- * The weak version was imported from cuML.
+ * The weak version (for undirected graphs, only) was imported from cuML.
  * This implementation comes from [1] and solves component labeling problem in
  * parallel on CSR-indexes based upon the vertex degree and adjacency graph.
  *
  * [1] Hawick, K.A et al, 2010. "Parallel graph component labelling with GPUs and CUDA"
+ * 
+ * The strong version (for directed or undirected graphs) is based on: 
+ * [2] Gilbert, J. et al, 2011. "Graph Algorithms in the Language of Linear Algebra"
+ *
+ * C = I | A | A^2 |...| A^k
+ * where matrix multiplication is via semi-ring: 
+ * (combine, reduce) == (&, |) (bitwise ops)
+ * Then: X = C & transpose(C); and finally, apply get_labels(X);
+ *
  *
  * @tparam IndexT the numeric type of non-floating point elements
  * @tparam TPB_X the threads to use per block when configuring the kernel
  * @param graph input graph; assumed undirected for weakly CC [in]
- * @param labels gdf_column for the output labels [out]
- * @param connectivity_type CUGRAPH_WEAK or CUGRAPH_STRONG
- * @param stream the cuda stream
+ * @param table of 2 gdf_columns: output labels and vertex indices [out]
+ * @param connectivity_type CUGRAPH_WEAK or CUGRAPH_STRONG [in]
+ * @param stream the cuda stream [in]
  */
 template<typename IndexT,
```

Comment:
The comment above (connectivity.cu from line 20 to 43) is redundant to the comment in cpp/include/algorithms.h. And AFAIK the standard way (in cugraph and in general as well) is to include this only in the header file. I haven't checked for every cugraph functions, but this is at least true for gdf_pagerank.

Reply:
There's nothing wrong w/ duplicating comments in headers and implementation files. It just makes information more readily available. AFAIK we didn't standardize the number of comments across source files. We cannot possibly standardize _everything_. Having _more_ comments cannot hurt. It helped the developer (me) when he wrote this; it will help the developers again when they need to update / add to the file.

Reply:
And in general it looks we need a file capturing all these "standardization" rules, many of which seem to be made up as we go. If there's a specific standardization please refer to the document stipulating it.

Reply:
OK, just a recommendation, not something deserves a debate.